bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/form/datetime/DatetimePickerConfig.java

```diff
@@ -0,0 +1,310 @@
+        return this;
+    }
+
+    /**
+     * Set maximum date.
+     *
+     * @param maxDate maximum date
+     * @return config instance
+     */
+    public DatetimePickerConfig withMaxDate(Date maxDate) {
+        put(MaxDate, customDateFormatter().format(maxDate));
+        return this;
+    }
+
+    /**
+     * Show today.
+     *
+     * @param showToday flag
+     * @return config instance
+     */
+    public DatetimePickerConfig setShowToday(Boolean showToday) {
+        put(ShowToday, showToday);
+        return this;
+    }
+
+    /**
+     * Set collapse.
+     *
+     * @param collapse flag
+     * @return config instance
+     */
+    public DatetimePickerConfig setCollapse(Boolean collapse) {
+        put(Collapse, collapse);
+        return this;
+    }
+
+    /**
+     * Set disabled
+     *
+     * @param disabledDates array disabled dates
+     * @return config instance
+     */
+    public DatetimePickerConfig withDisabledDates(Date[] disabledDates) {
+        put(DisabledDates, convertDatesToStrings(disabledDates));
+        return this;
+    }
+
+    /**
+     * Set enabled
+     *
+     * @param enabledDates array enabled dates
+     * @return config instance
+     */
+    public DatetimePickerConfig withEnabledDates(Date[] enabledDates) {
+        put(EnabledDates, convertDatesToStrings(enabledDates));
+        return this;
+    }
+
+    /**
+     * Get date format.
+     *
+     * @return date format
+     */
+    public String getFormat() {
+        return toJavaDateFormat(getString(Format));
+    }
+
+    /**
+     * Get minute stepping
+     *
+     * @return minute stepping
+     */
+    public String getMinuteStepping() {
+        return getString(MinuteStepping);
+    }
+
+    /**
+     * Get minute stepping
+     *
+     * @return minute stepping
+     */
+    public ViewModeType getViewMode() {
+        return ViewModeType.valueOf(getString(ViewMode));
+    }
+
+    /**
+     * Convert array dates to array strings.
+     *
+     * @param dates array
+     * @return array strings
+     */
+    private String[] convertDatesToStrings(Date[] dates) {
+        String[] stringDates = new String[dates.length];
+        SimpleDateFormat formatter = customDateFormatter();
+        int i = 0;
+        for (Date date : dates) {
+            stringDates[i++] = formatter.format(date);
+        }
+        return stringDates;
+    }
+
+    /**
+     * Get date formatter based config format.
+     *
+     * @return date formatter
+     */
+    private SimpleDateFormat customDateFormatter() {
+        return new SimpleDateFormat(getFormat());
+    }
+
+    /**
+     * Convert java script date format to java.
+     *
+     * @param javaScriptDateFormat date format
+     * @return java date format
+     */
+    public static String toJavaDateFormat(final String javaScriptDateFormat) {
+        return nullToEmpty(javaScriptDateFormat).replaceAll("DD", "dd").replaceAll("YYYY", "yyyy");
```

Comment:
Isn't it better to replace (all) `Y` to `y` ?
This way it will work for `YY` too.